core/lib/zksync_core/src/consensus/mod.rs

```diff
@@ -1,6 +1,62 @@
 //! Consensus-related functionality.
+use std::sync::Arc;
+
+use anyhow::Context as _;
+use zksync_concurrency::{ctx, scope};
+use zksync_consensus_executor::{ConsensusConfig, Executor, ExecutorConfig};
+use zksync_consensus_roles::{node, validator};
+use zksync_dal::ConnectionPool;
+use zksync_types::Address;
 
 mod payload;
 mod proto;
+mod storage;
+
+#[cfg(test)]
+pub(crate) mod testonly;
+#[cfg(test)]
+mod tests;
+
+pub(crate) use self::{payload::Payload, storage::sync_block_to_consensus_block};
+
+pub struct Config {
+    pub executor: ExecutorConfig,
+    pub consensus: ConsensusConfig,
+    pub node_key: node::SecretKey,
+    pub validator_key: validator::SecretKey,
+    pub operator_address: Address,
+}
 
-pub(crate) use self::payload::Payload;
+impl Config {
+    #[allow(dead_code)]
+    pub async fn run(self, ctx: &ctx::Ctx, pool: ConnectionPool) -> anyhow::Result<()> {
+        assert_eq!(
```

Comment:
Nit: Given that we propagate errors here, should this use `anyhow::ensure!` instead?

Reply:
done